Update README to include information on `buildifier` option

luispadron opened the pull request:
## Summary

Updates the `README.md` file to describe the `buildifier` option and how to use it.

jerrymarino commented on lib/cocoapods/bazel.rb:
```diff
@@ -135,18 +135,15 @@ def self.format_files(build_files:, buildifier:, workspace:)
-      args += %w[-type build]
 
       executable, *args = args
-      Pod::Executable.execute_command executable,
```
It looked like it results in different behavior when specifying buildifier as an array vs true?
1. `true`: format only governed BUILD files
2. `/path/to/buildifier`: formats the entire directory: no way to specify `cocoapods-bazel` only ones.

Consider not moving the part of the command `.each_key.map { |d| File.join workspace, d, 'BUILD.bazel' }` if so, this seems hard to reason about. I'd suggest to align them to do the same thing. It's nice that it doesn't try to mutate other files today.

I assumed what you wanted to do is pass `bazel run :buildifier --` - the key bit adding `--` so bazel passes arguments to the binary vs formatting the entire repo

luispadron replied:
🙏🏼  `--` 🙏🏼

jerrymarino commented on lib/cocoapods/bazel.rb:
```diff
@@ -135,18 +135,15 @@ def self.format_files(build_files:, buildifier:, workspace:)
-        args = ['buildifier']
+        args = %w[buildifier -type build] + build_files.each_key.map { |d| File.join workspace, d, 'BUILD.bazel' }
       when String, Array
         args = Array(buildifier)
```
This line originally accepted an arbitrary array for  `buildifer`: e.g. today, can't you pass `bazel run :buildifier` Mainly it sets the `executable` to that below

luispadron replied:
Undid this, totally right!

jerrymarino commented on README.md:
```diff
@@ -50,6 +50,14 @@ There are some experimental features that are opt-in and can be enabled adding t
+
+You can disable buildifier formatting with `buildifier: false` in the options of the `cocoapods-bazel` plugin.
+
+Additionally, if you'd like to use a custom `buildifier` executable you can provide the `cocoapods-bazel` plugin options with an array of arguments to execute to format files. For example, if you have `buildifier` runnable target you've defined in Bazel with the name `buildifier` you can run this specific version with: `buildifier: ['bazel', 'run', 'buildifier']`.
```
Does this already work today on without this change? It looks like the code works in this way but we popped off the default arguments - you'd just need to add `--` to the array here

luispadron replied:
Hey, you're totally right. I definitely tried the `--` locally but for some reason didn't think it would work because the `builidifer` rule with different mode threw me off. I just verified via using the plugin that appending `--` allows the arguments and build file formatting

jerrymarino reviewed (commented):
It looks like the code is already here to run `bazlisk run buildifier` or `/path/to/bazelisk run :buildifier` - today if I set:
```
buildifer: [bazelisk run buildifier --]
```
it just works? e.g. today you can have a version controlled and bazel managed buildifier as-is

luispadron commented:
🤦🏼 @jerrymarino Thanks for the help here, you're right that this is totally still supported right now and in fact works better (provides only the cocoapods-bazel) files to format so we should keep this code as-is.


If we want to pull in the README update we can still merge this PR but I've removed all the actual code changes. I updated the README to make it more clear how this option works for others.
